app/forms/upload_form.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class UploadForm
+  include ActiveModel::Validations
+
+  attr_reader :search_engine, :csv_file
+
+  validates :search_engine, presence: true
+  validates :search_engine, inclusion: { in: SearchResult.search_engines.keys }
+  validates :csv_file, presence: true
+
+  validate :valid_content_type, if: -> { csv_file.present? }
+  validate :valid_count, if: -> { csv_file.present? }
+
+  def initialize(params)
+    @search_engine = params[:search_engine]
+    @csv_file = params[:csv_file]
+  end
+
+  def valid_content_type
+    return if csv_file.content_type == 'text/csv'
+
+    add_error :csv_file_invalid_extension
+  end
+
+  def valid_count
+    return if CSV.read(csv_file.tempfile).count.between?(1, 1000)
+
+    add_error :csv_file_invalid_count
+  end
+
+  def add_error(type)
+    errors.add(:base, I18n.t("upload_form.validation.#{type}"))
+  end
+
+  def save
+    content = csv_file.read
+    keywords = CSV.parse(content).flatten.map(&:strip)
+
+    search_results = keywords.map { |keyword| SearchResult.new(keyword: keyword, search_engine: search_engine) }
+    SearchResult.import search_results
```

Comment:
Why using import over just the save method each keyword result?

Reply:
From my understanding, this approach is more performant

Ex: 100 keywords, instead of doing 100 database transactions, we only do one transaction 🚀

Reply:
You are right but prefer to use the built-in [insert_all](https://apidock.com/rails/v6.0.0/ActiveRecord/Persistence/ClassMethods/insert_all) method provided by ActiveRecord

Reply:
Oh, I initially did, but rubocop gave a warning about it, should I still proceed? 💭

Reply:
Oh, I initially did, but rubocop gave a warning about it, should I still proceed? 💭

Reply:
I'll give it another try

Reply:
Oh I see then that should be okay to move forward with the current implementation